code/modules/antagonists/_common/antag_spawner.dm

```diff
@@ -281,3 +296,58 @@
 	M.mind.objectives += KillDaCrew
 	to_chat(M, "<B>Objective #[1]</B>: [KillDaWiz.explanation_text]")
 	to_chat(M, "<B>Objective #[2]</B>: [KillDaCrew.explanation_text]")
+
+///////////Revenant
+
+/obj/item/antag_spawner/revenant
+	name = "vial of ectoplasm"
+	desc = "A magically infused bottle of ectoplasm, effectivly pure salt from the spectral realm."
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "vialectoplasm"
+	var/shatter_msg = "<span class='notice'>You shatter the bottle, no \
+		turning back now!</span>"
+	var/veil_msg = "<span class='warning'>The ectoplasm is awake and seeps \
+		away...</span>"
+	var/objective_verb = "Harvest"
+	var/mob/living/revenant = /mob/living/simple_animal/revenant
+
+/obj/item/antag_spawner/revenant/attack_self(mob/user)
+	if(level_blocks_magic(user.z)) //this is to make sure the wizard does NOT summon a revenant from the Den..
```

Comment:
This all should prob use a refactor since the stuff below should be quite similar to most other `antag_spawner`s.
Other PR though